home_assistant_streamdeck_yaml.py

```diff
@@ -62,9 +64,281 @@
+            if isinstance(dial.icon, str) and ":" in dial.icon:
+                which, id_ = dial.icon.split(":",1)
+                if which == "spotify":
+                    filename = _to_filename(dial.icon, ".jpeg")
+                    image = _download_spotify_image(id_,filename).copy()
+                if which == "url":
+                    filename = _url_to_filename(id_)
+                    image = _download_image(id_, filename, size).copy()
+                if which == "ring":
+                    pct = _maybe_number(id_)
+                    assert isinstance(pct, (int, float)), f"Invalid ring percentage: {id_}"
+                    image = _draw_percentage_ring(
+                        percentage=pct, 
+                        size=size, 
+                        radius = 40,
+                    )
+            
+            icon_convert_to_grayscale = False
+            text = dial.text
+            text_color = dial.text_color or "white"
+            
+            if complete_state[dial.entity_id]["state"] == "off" and dial.icon_gray_when_off == True:
+                icon_convert_to_grayscale = True
+                
+            
+            if image is None:
+                image = _init_icon(
+                    icon_background_color = dial.icon_background_color,
+                    icon_filename = dial.icon,
+                    icon_mdi = dial.icon_mdi,
+                    icon_mdi_margin=icon_mdi_margin,
+                    icon_mdi_color=_named_to_hex(dial.icon_mdi_color or text_color),
+                    size=size
+                ).copy()
+            
+            if icon_convert_to_grayscale:
+                image = _convert_to_grayscale(image)
+            
+            _add_text(
+                image=image,
+                font_filename=font_filename,
+                text_size=self.text_size,
+                text=text,
+                text_color=text_color,
+                text_offset=self.text_offset,
+            )
+            return image
+        
+        except Exception as e:
+            console.log(e)
+            warnings.warn(
+                f"Failed to render icon for dial {key}",
+                IconWarning,
+                stacklevel=2,
+            )
+            return _generate_failed_icon(
+                size=size
+            )
+    
+    _timer: AsyncDelayedCallback | None = PrivateAttr(None)
+    
+    def start_or_restart_timer(
+        self,
+        callback: Callable[[], None | Coroutine] | None = None    
+    ) -> bool:
+        if self.delay:
+            if self._timer is None:
+                assert isinstance(
+                    self.delay,
+                    (int, float),
+                ), f"Invalid delay: {self.delay}"
+                self._timer = AsyncDelayedCallback(delay=self.delay, callback=callback)
+            if self._timer.is_running():
+                self._timer.cancel()
+                self._timer.start()
+            else:
```

Comment:
timer is always started, move `self._timer.start()` out of the if block and remove the else block.